apps/antalmanac/src/components/RightPane/CoursePane/SearchForm/RestrictionsFilter.tsx

```diff
@@ -0,0 +1,183 @@
+
+const restrictionList = [
+    // "All: Include All Restrictions",
+    // "NONE: Filter all restrictions",
+    'A: Prerequisite required',
+    'B: Authorization code required',
+    'C: Fee required',
+    'D: Pass/Not Pass option only',
+    'E: Freshmen only',
+    'F: Sophomores only',
+    'G: Lower-division only',
+    'H: Juniors only',
+    'I: Seniors only',
+    'J: Upper-division only',
+    'K: Graduate only',
+    'L: Major only',
+    'M: Non-major only',
+    'N: School major only',
+    'O: Non-school major only',
+    'R: Biomedical Pass/Fail course (School of Medicine only)',
+    'S: Satisfactory/Unsatisfactory only',
+    'X: Separate authorization codes required to add, drop, or change enrollment',
+];
+
+// clean up styling later
+const styles = {
+    formControl: {
+        flexGrow: 1,
+        marginRight: 15,
+        width: 120,
+    },
+};
+
+// this typecasting is pretty nasty, but it won't work without it
+// If there's a more proper workaround please fix/LMK
+const staticRestrictionList = {
+    [String('A')]: 'A: Prerequisite required',
+    [String('M')]: 'M: Non-major only',
+    [String('E')]: 'E: Freshmen only',
+    [String('G')]: 'G: Lower-division only',
+    [String('I')]: 'I: Seniors only',
+    [String('N')]: 'N: School major only',
+    [String('F')]: 'F: Sophomores only',
+    [String('O')]: 'O: Non-school major only',
+    [String('H')]: 'H: Juniors only',
+    [String('J')]: 'J: Upper-division only',
+    [String('C')]: 'C: Fee required',
+    [String('D')]: 'D: Pass/Not Pass option only',
+    [String('X')]: 'X: Separate authorization codes required to add, drop, or change enrollment',
+    [String('R')]: 'R: Biomedical Pass/Fail course (School of Medicine only)',
+    [String('K')]: 'K: Graduate only',
+    [String('S')]: 'S: Satisfactory/Unsatisfactory only',
+    [String('B')]: 'B: Authorization code required',
+    [String('L')]: 'L: Major only',
+};
+
+interface RestrictionFilterProps {
+    classes: ClassNameMap;
+}
+
+interface RestrictionFilterState {
+    restrictions: string[];
+}
+
+// url is not being grabbed to set the values correctly on refresh
+class RestrictionsFilter extends PureComponent<RestrictionFilterProps, RestrictionFilterState> {
+    updateRestrictionsAndGetFormData() {
+        RightPaneStore.updateFormValue('restrictions', RightPaneStore.getUrlRestrictionsValue());
+        return RightPaneStore.getFormData().restrictions;
+    }
+
+    getRestrictions() {
+        return RightPaneStore.getUrlRestrictionsValue().trim()
+            ? this.updateRestrictionsAndGetFormData()
+            : RightPaneStore.getFormData().restrictions;
+    }
+
+    state = {
+        restrictions:
+            this.getRestrictions() !== 'ALL' && typeof this.getRestrictions() === 'string' // guards for type errors
+                ? this.getRestrictions()
+                      .split('')
+                      .filter((item: any) => !Array.isArray(item))
```

Comment:
```suggestion
                      .filter((item: unknown): item is string => !Array.isArray(item))
```
Filters can narrow types if you use a type-guard